src/catch2/reporters/catch_reporter_listening.cpp

```diff
@@ -15,151 +17,157 @@ namespace Catch {
             listener->noMatchingTestCases( spec );
         }
-        m_reporter->noMatchingTestCases( spec );
     }
 
     void ListeningReporter::reportInvalidArguments(std::string const&arg){
         for ( auto& listener : m_listeners ) {
             listener->reportInvalidArguments( arg );
         }
-        m_reporter->reportInvalidArguments( arg );
     }
 
     void ListeningReporter::benchmarkPreparing( StringRef name ) {
         for (auto& listener : m_listeners) {
             listener->benchmarkPreparing(name);
         }
-        m_reporter->benchmarkPreparing(name);
     }
     void ListeningReporter::benchmarkStarting( BenchmarkInfo const& benchmarkInfo ) {
         for ( auto& listener : m_listeners ) {
             listener->benchmarkStarting( benchmarkInfo );
         }
-        m_reporter->benchmarkStarting( benchmarkInfo );
     }
     void ListeningReporter::benchmarkEnded( BenchmarkStats<> const& benchmarkStats ) {
         for ( auto& listener : m_listeners ) {
             listener->benchmarkEnded( benchmarkStats );
         }
-        m_reporter->benchmarkEnded( benchmarkStats );
     }
 
     void ListeningReporter::benchmarkFailed( StringRef error ) {
         for (auto& listener : m_listeners) {
             listener->benchmarkFailed(error);
         }
-        m_reporter->benchmarkFailed(error);
     }
 
     void ListeningReporter::testRunStarting( TestRunInfo const& testRunInfo ) {
         for ( auto& listener : m_listeners ) {
             listener->testRunStarting( testRunInfo );
         }
-        m_reporter->testRunStarting( testRunInfo );
     }
 
     void ListeningReporter::testGroupStarting( GroupInfo const& groupInfo ) {
         for ( auto& listener : m_listeners ) {
             listener->testGroupStarting( groupInfo );
         }
-        m_reporter->testGroupStarting( groupInfo );
     }
 
 
     void ListeningReporter::testCaseStarting( TestCaseInfo const& testInfo ) {
         for ( auto& listener : m_listeners ) {
             listener->testCaseStarting( testInfo );
         }
-        m_reporter->testCaseStarting( testInfo );
     }
 
     void ListeningReporter::sectionStarting( SectionInfo const& sectionInfo ) {
         for ( auto& listener : m_listeners ) {
             listener->sectionStarting( sectionInfo );
         }
-        m_reporter->sectionStarting( sectionInfo );
     }
 
     void ListeningReporter::assertionStarting( AssertionInfo const& assertionInfo ) {
         for ( auto& listener : m_listeners ) {
             listener->assertionStarting( assertionInfo );
         }
-        m_reporter->assertionStarting( assertionInfo );
     }
 
     // The return value indicates if the messages buffer should be cleared:
     bool ListeningReporter::assertionEnded( AssertionStats const& assertionStats ) {
-        for( auto& listener : m_listeners ) {
-            static_cast<void>( listener->assertionEnded( assertionStats ) );
+        const bool alwaysReport =
```

Comment:
Can we have different name for this? I remember this name being in more places but I think it sends the wrong message.